### PR TITLE
Changes to usage of DrupalFinderComposerRuntime API in ScriptHandler

### PR DESCRIPTION
**Pull request by jnoordsij**
- Add sanity null-check for `$drupalRoot`
- Restore `Path` generator for `config_sync_directory`

This is a rebased version of #650 that reverts some changes introduced in #673.
Hopefully due to the changes introduced in https://github.com/webflo/drupal-finder/pull/73 and the subsequent release (https://github.com/webflo/drupal-finder/releases/tag/1.3.1), the CI should now pass without manual overriding of the config sync directory.

This probably solves #675.

Note: if preferable, I could update `composer.json` to explicitly require `"webflo/drupal-finder": "^1.3.1"`.

### CODE REVIEW

File: scripts/composer/ScriptHandler.php [hunk 1]
```diff
@@ -22,6 +22,15 @@ public static function createRequiredFiles(Event $event) {
     $drupalFinder = new DrupalFinderComposerRuntime();
     $drupalRoot = $drupalFinder->getDrupalRoot();
 
+    // If Drupal root was not found, exit.
```
Comment (AlexSkrypnyk) [inline scripts/composer/ScriptHandler.php]: nit: follow the same code style as the rest of the file
```suggestion
    if (is_null($drupalRoot)) {
      $event->getIO()->writeError('<error>Drupal root could not be detected.</error>');
      exit(1);
    }
```

**Review by AlexSkrypnyk — CHANGES_REQUESTED** [review]
@jnoordsij 
Thank you for your contribution!

I think this looks good. Could you please update the code to follow the code style used in the current file.

File: scripts/composer/ScriptHandler.php (hunk 1 above)
Comment (jnoordsij) [reply scripts/composer/ScriptHandler.php]: Fair point! I followed more or less the calls from `checkComposerVersion`, but this looks much cleaner!

**Comment by webflo** [comment]
👍

**Review by normanlolx — APPROVED** [review]

**Comment by normanlolx** [comment]
Glad to see this sorted out dynamically. 🤩

**Review by AlexSkrypnyk — APPROVED** [review]

**Pull request merged**